influence_benchmark/backend/hf_backend.py

```diff
@@ -3,29 +3,41 @@
 
 import torch
 import torch.nn.functional as F
+from peft import LoraConfig, TaskType
 from transformers import AutoModelForCausalLM, AutoTokenizer
 
 from influence_benchmark.backend.backend import Backend
 
 
 class HFBackend(Backend):
-    """A multiton class for managing multiple instances of the Hugging Face backend. This class is a singleton for each model name.
-    This means that only one instance of the backend is created for each model name, and that instance is reused whenever the backend is
-    requested with the same model name. This reduces the memory usage of the backend.
-    """
 
     def __init__(self, model_name, device, lora_path=None):
-        self.model = AutoModelForCausalLM.from_pretrained(model_name).half().eval().to(device)
+        config = LoraConfig(
+            task_type=TaskType.CAUSAL_LM,
+            r=8,
```

Comment:
Maybe worth trying different values of this at some point (and these other hyperparams), or putting it into a config somewhere to be able to more easily sweep it